### ISSUE TITLE
As<TInterface> strange behavior

### ISSUE THREAD

**Issue by Belorus** [moq/moq4#458]
The following code fails
```
Mock<IService2> mock = new Mock<TestFriendlyService>().As<IService>();

object o =    ((Mock)mock).Object;
IService2 o2 =       mock .Object;

Assert.True(object.ReferenceEquals(o, o2));
```

Is that ok ?

**Comment by stakx** [comment]
That *is* a bit odd. I'll take a look at it later today.

**Comment by stakx** [comment]
Looks like a mock "cast" to a different type via `.As<T>()` doesn't properly keep track of its underlying type, I'd say that's a bug.

**Comment by stakx** [comment]
@Belorus, this will be fixed in the next release of Moq (version >4.7.127).
